blocks/main_loop.py

```diff
@@ -156,16 +159,17 @@ def _check_finish_training(self):
         # the iteration the corresponding log record can be found only in
         # the previous row.
         if (self.log.current_row.training_finish_requested or
-                self.log.current_row.keyboard_interrupt_received or
-                self.log.previous_row.keyboard_interrupt_received):
+                self.log.current_row.interrupt_received or
+                self.log.previous_row.interrupt_received):
             raise TrainingFinish
 
-    def _handle_keyboard_interrupt(self, signal_number, frame):
+    def _handle_interrupt(self, signal_number, frame):
         # After receiving a first keyboard interrupt signal,
         # ignore all following ones.
```

Comment:
That's what I meant, actually. I think we should just pickle to `.model.pkl.tmp` or something, and once pickling has completed successfully, we move `.model.pkl.tmp` to `model.pkl`. There will always be a split second in the middle of copying in which an interruption might happen, but at least we will never corrupt an good copy because of recursion errors, pickling errors, etc.